pymc_experimental/utils/model_fgraph.py

```diff
@@ -0,0 +1,174 @@
+from typing import Optional, Sequence
+
+import pytensor
+from pymc.model import Model
+from pytensor.graph import Apply, FunctionGraph, Op
+from pytensor.tensor import TensorVariable
+
+from pymc_experimental.utils.pytensorf import StringType
+
+
+class ModelVar(Op):
```

Comment:
My main concern is that you are wrapping every random variable with variants of this `Op`. That might make it harder for rewrites to work. For example, imagine a model that uses the non centered parametrization and you want to rewrite it into the centered parametrization. You would need to look at many things in the rewrite: whether the two entities that are being summed together are `FreeRV`s or if its a single `FreeRV` plus some product of another thing. This extra `Op` layer, in my opinion, complicates the operation of the rewrites.

From what I understood, the main advantage of these new `Op`s is that they track two things: the kind of variable within the model, and the dimension information of the variable. I know that it's a bit of a crappy thing to ask, but have you considered using `tag`s to keep track of the variable kind instead of `Op`s?

Reply:
> whether the two entities that are being summed together are FreeRVs or if its a single FreeRV plus some product of another thing. This extra Op layer, in my opinion, complicates the operation of the rewrites.

I responded in the discussion. I think we want to have to look at what is a `FreeRV` and what's not, because you want to explicitly state what old `FreeRV` is to be replaced by the new centered one (that is move the value_variable one node lower). I don't think the rewrite is any more hard, the rule would be:
```
shift + FreeRV(Normal(mu, sigma)) * scale  -> FreeRV(Normal(mu + shift, sigma * scale)
```
A bit of a forced case, but you wouldn't want this rewrite to do this, on top of the following
```
shift + ObservedRV(Normal(mu, sigma) * scale
```
So you would always need to inspect the Value variable type anyway. Then you might have it in the graph already.